### DISCUSSION TITLE
coordinates as multiple filters

### DISCUSSION THREAD

**Discussion by sagregrevc** [pydata/xarray#8158] [Q&A]
Which is right way filter data if I have less **bold coordinates** than needed filters?
![image](https://github.com/pydata/xarray/assets/115998253/2ce96fb9-6afa-40d8-8069-931d8ccdb4db)

somehow hvplot can understand what I want. but xarray not (I need change data later with xarray, before plotting)
this command work well:
`combined.hvplot(x="pixels", y="int", by="exposure")`
but this not:
`da.isel(exposure=1.0)`
"ValueError: Dimensions {'exposure'} do not exist. Expected one or more of Frozen({'pixels': 2048})"

```
spectra_s = []

for filename in [f for f in filename_s if f.endswith(".xml")]:
    xml_file_path = f"./raw/_/o1/{filename}"

    df, df_fil, df_metadata = take_df(xml_file_path)
    
    grating = df_metadata.loc[11,"C"]
    prototype = df_metadata.loc[10,"C"]
    light_source = df_metadata.loc[29,"C"]
    exposure = float(df_metadata.loc[23,"C"])
    
    da = xr.DataArray(data=df.loc[:, "int"], 
                      coords={"grating": grating, 
                              "lamp":light_source,
                              "exposure":exposure},
                      dims=["pixels"], )
    
    spectra_s.append(da)

combined = xr.concat(spectra_s, dim="exposure")
combined
```

**Comment by sagregrevc** [comment]
tried with Dataset. 
can find "exposure". but cannot find other 2

**Reply by dcherian** [reply]
You'll need to. set these additional variables. as "indexes" using [set_xindex](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.set_xindex.html)

**Reply by Xaego** [reply]
thank you. this helpful command, but I meet another problem
now I have dataset
![image](https://github.com/pydata/xarray/assets/34896747/b6cc1e2d-0912-4504-83f1-d1571414cef5)

ValueError: cannot re-index or align objects with conflicting indexes found for the following dimensions: 'exposure' (2 conflicting indexes)
Conflicting indexes may occur when
- they relate to different sets of coordinate and/or dimension names
- they don't have the same type
- they may be used to reindex data along common dimensions


```
folder_xml_files = "./raw"
folder_xml_path = f"./{folder_xml_files}/"

global_spectra_s = []

for dirpath, _, filename_s in tqdm(os.walk(folder_xml_path), position=0, leave=True):
    spectra_s = []
    
    for filename in [f for f in filename_s if f.endswith(".xml")]:
        xml_file_path = f"{dirpath}/{filename}"
    
        df, df_fil, df_metadata = take_df(xml_file_path)
        
        grating = df_metadata.loc[11,"C"]
        prototype = df_metadata.loc[10,"C"]
        light_source = df_metadata.loc[29,"C"]
        exposure = float(df_metadata.loc[23,"C"])
        
        ds = xr.Dataset(
            data_vars={"int": (["pixels"], df.loc[:, "int"])},
            coords={"grating": grating, 
                  "lamp":light_source,
                  "exposure":exposure},
        )
        
        spectra_s.append(ds)
    
    if spectra_s:
        ds_device = xr.concat(spectra_s, dim="exposure")
        global_spectra_s.append(ds_device)
    
if global_spectra_s:
    final_dataset = xr.concat(global_spectra_s, dim="grating")
final_dataset = final_dataset.set_xindex(coord_names=["lamp"])
    
final_dataset.hvplot(x="pixels", y="int", by="exposure", width=1500, height=1000)
```
PS I same person

**Reply by Xaego** [reply]
`join="override" `seems solved problem
`ds_device = xr.concat(spectra_s, dim="exposure", join="override")`
